services/api/mw2pdf/src/classes/MediaWikiSession.ts

```diff
@@ -1,131 +1,219 @@
 import puppeteer from 'puppeteer'
 import {
-  PdfGenerator
-} from './PdfGenerator'
+  PdfFactory
+} from './PdfFactory'
 import {
   makeLoginRequest,
   makeLoginTokenRequest,
   extractLoginToken,
 } from '../utils/mediaWiki'
 import {
   parseCookies,
-  toCookieString,
+  ParsedCookie,
 } from '../utils/cookies'
 import {
   deletePdfs,
 } from '../utils/files'
+import { 
+  Pdf,
+  PdfConstructorOptions
+} from './Pdf'
+import { v4 } from 'uuid';
+import { join } from 'path';
+
+/**
+ * MwPdfOptions
+ * 
+ * A class to hold the options that control creation of a PDF (booklet)
+ * from a MediaWiki site.
+ */
+export class MwPdfOptions {
```

Comment:
Can this be turned into an Interface instead of a Class?

Reply:
Absolutely! And it is!